src/main/java/net/dv8tion/jda/core/requests/restaction/GuildAction.java

```diff
@@ -0,0 +1,733 @@
+     *         to use for the construction of the Channel
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided channel is {@code null}!
+     *
+     * @return The current GuildAction for chaining convenience
+     */
+    @CheckReturnValue
+    public GuildAction addChannel(ChannelData channel)
+    {
+        Checks.notNull(channel, "Channel");
+        this.channels.add(channel);
+        return this;
+    }
+
+    /**
+     * Gets the {@link net.dv8tion.jda.core.requests.restaction.GuildAction.ChannelData ChannelData}
+     * of the specified index. The index is 0 based on insertion order of {@link #addChannel(GuildAction.ChannelData)}!
+     *
+     * @param  index
+     *         The 0 based index of the channel
+     *
+     * @throws java.lang.IndexOutOfBoundsException
+     *         If the provided index is not in bounds
+     *
+     * @return The current GuildAction for chaining convenience
+     */
+    @CheckReturnValue
+    public ChannelData getChannel(int index)
+    {
+        return this.channels.get(index);
+    }
+
+    // Roles
+
+    /**
+     * Retrieves the {@link net.dv8tion.jda.core.requests.restaction.GuildAction.RoleData RoleData} for the
+     * public role ({@link net.dv8tion.jda.core.entities.Guild#getPublicRole() Guild.getPublicRole()}) for the resulting Guild.
+     * <br>The public role is also known in the official client as the {@code @everyone} role.
+     *
+     * <p><b>You can only change the permissions of the public role!</b>
+     *
+     * @return RoleData of the public role
+     */
+    @CheckReturnValue
+    public RoleData getPublicRole()
+    {
+        return this.roles.get(0);
+    }
+
+    /**
+     * Retrieves the {@link net.dv8tion.jda.core.requests.restaction.GuildAction.RoleData RoleData} for the
+     * provided index.
+     * <br>The public role is at the index 0 and all others are ordered by insertion order!
+     *
+     * @param  index
+     *         The index of the role
+     *
+     * @throws java.lang.IndexOutOfBoundsException
+     *         If the provided index is out of bounds
+     *
+     * @return RoleData of the provided index
+     */
+    @CheckReturnValue
+    public RoleData getRole(int index)
+    {
+        return this.roles.get(index);
+    }
+
+    /**
+     * Creates and add a new {@link net.dv8tion.jda.core.requests.restaction.GuildAction.RoleData RoleData} object
+     * representing a Role for the resulting Guild.
+     *
+     * <p>This can be used in {@link GuildAction.ChannelData#addPermissionOverride(GuildAction.RoleData, long, long) ChannelData.addPermissionOverride(...)}.
+     * <br>You may change any properties of this {@link net.dv8tion.jda.core.requests.restaction.GuildAction.RoleData RoleData} instance!
+     *
+     * @return RoleData for the new Role
+     */
+    @CheckReturnValue
+    public RoleData newRole()
```

Comment:
The difference between how one adds a new role vs a new channel feels odd.

Consider, at minimum, providing a `newChannel(ChannelType, name: String)`